src/emitters/emitter.sqns.ts

```diff
@@ -717,6 +718,20 @@ export class SqnsEmitter implements IEmitter {
     }
   }
 
+  private getSeparateConsumer(consumer: Pick<Topic, 'separateConsumerGroup' | 'consumerGroup'>): string | undefined {
+    const { separateConsumerGroup, consumerGroup } = consumer;
+    if (separateConsumerGroup && consumerGroup)
+      throw new Error(
+        `separateConsumerGroup and consumerGroup cannot be used together`
+      );
+    return separateConsumerGroup || consumerGroup?.name;
+  }
+
+  private checkIfConsumerIsFifo(topic: Pick<Topic, 'consumerGroup' | 'isFifo'>): boolean {
```

Comment:
no need to pick here as we are passing topic anyways

Reply:
done.